src/wp-includes/blocks/template-part.php

```diff
@@ -0,0 +1,155 @@
+			if ( 0 === validate_file( $attributes['slug'] ) && file_exists( $template_part_file_path ) ) {
+				$content = _inject_theme_attribute_in_block_template_content( file_get_contents( $template_part_file_path ) );
+			}
+		}
+	}
+
+	if ( is_null( $content ) && is_user_logged_in() ) {
+		if ( ! isset( $attributes['slug'] ) ) {
+			// If there is no slug this is a placeholder and we dont want to return any message.
+			return;
+		}
+		return sprintf(
+			/* translators: %s: Template part slug. */
+			__( 'Template part has been deleted or is unavailable: %s' ),
+			$attributes['slug']
+		);
+	}
+
+	if ( isset( $seen_ids[ $template_part_id ] ) ) {
+		// WP_DEBUG_DISPLAY must only be honored when WP_DEBUG. This precedent
+		// is set in `wp_debug_mode()`.
+		$is_debug = defined( 'WP_DEBUG' ) && WP_DEBUG &&
+			defined( 'WP_DEBUG_DISPLAY' ) && WP_DEBUG_DISPLAY;
+
+		return $is_debug ?
+			// translators: Visible only in the front end, this warning takes the place of a faulty block.
+			__( '[block rendering halted]' ) :
+			'';
+	}
+
+	// Run through the actions that are typically taken on the_content.
+	$seen_ids[ $template_part_id ] = true;
+	$content                       = do_blocks( $content );
+	unset( $seen_ids[ $template_part_id ] );
+	$content = wptexturize( $content );
+	$content = convert_smilies( $content );
+	$content = shortcode_unautop( $content );
+	if ( function_exists( 'wp_filter_content_tags' ) ) {
+		$content = wp_filter_content_tags( $content );
+	} else {
+		$content = wp_make_content_images_responsive( $content );
+	}
+	$content = do_shortcode( $content );
+
+	if ( empty( $attributes['tagName'] ) ) {
+		$defined_areas = get_allowed_block_template_part_areas();
+		$area_tag      = 'div';
+		foreach ( $defined_areas as $defined_area ) {
+			if ( $defined_area['area'] === $area && isset( $defined_area['area_tag'] ) ) {
+				$area_tag = $defined_area['area_tag'];
+			}
+		}
+		$html_tag = $area_tag;
+	} else {
+		$html_tag = esc_attr( $attributes['tagName'] );
+	}
+	$wrapper_attributes = get_block_wrapper_attributes();
+
+	return "<$html_tag $wrapper_attributes>" . str_replace( ']]>', ']]&gt;', $content ) . "</$html_tag>";
+}
+
+/**
+ * Returns an array of variation objects for the template part block.
+ *
+ * @return array Array containing the block variation objects.
+ */
+function build_template_part_block_variations() {
+	$variations    = array();
+	$defined_areas = get_allowed_block_template_part_areas();
+	foreach ( $defined_areas as $area ) {
+		if ( 'uncategorized' !== $area['area'] ) {
+			$variations[] = array(
+				'name'        => $area['area'],
+				'title'       => $area['label'],
+				'description' => $area['description'],
+				'attributes'  => array(
+					'area' => $area['area'],
+				),
+				'scope'       => array( 'inserter' ),
+				'icon'        => $area['icon'],
+			);
+		}
+	}
+	return $variations;
+}
+
+/**
+ * Registers the `core/template-part` block on the server.
+ */
```

Comment:
```suggestion
/**
 * Registers the `core/template-part` block on the server.
 *
 * @since 5.9.0
 */
```

Reply:
This will need to be changed in Gutenberg as this file is copied from `@wordpress/block-library`.

Reply:
As stated elsewhere, it looks like dynamic blocks PHP shared from GB doesn't typically contain `@since` PHPDocs (as the version would be ambiguous, depending on whether it's used in a GB or Core context).